aten/src/ATen/native/mps/operations/GridSampler.mm

```diff
@@ -0,0 +1,125 @@
+
+namespace at {
+namespace native {
+
+void grid_sampler_2d_mps_impl(Tensor &output, const Tensor& input, const Tensor& grid,
+                              int64_t interpolation_mode, int64_t padding_mode,
+                              bool align_corners) {
+  using namespace mps;
+  check_grid_sampler_common(input, grid);
+  check_grid_sampler_2d(input, grid);
+
+  MPSGraphResizeMode samplingMode;
+  MPSGraphPaddingMode paddingMode;
+
+  auto memory_format = input.suggest_memory_format();
+  MPSGraphTensorNamedDataLayout inputTensorLayout =
+      (memory_format == at::MemoryFormat::Contiguous) ? MPSGraphTensorNamedDataLayoutNCHW : MPSGraphTensorNamedDataLayoutNHWC;
+
+  switch (static_cast<GridSamplerPadding>(padding_mode)) {
+    case GridSamplerPadding::Zeros:
+      paddingMode = MPSGraphPaddingModeZero; break;
+    case GridSamplerPadding::Border:
+      TORCH_CHECK(false, "MPS: Unsupported Border padding mode"); break;
+    case GridSamplerPadding::Reflection:
+      paddingMode = MPSGraphPaddingModeReflect; break;
+    default:
+      TORCH_CHECK(false, "MPS: Unrecognised Padding Mode: ", padding_mode);
+  }
+
+  switch (static_cast<GridSamplerInterpolation>(interpolation_mode)) {
+    case GridSamplerInterpolation::Bilinear:
+      samplingMode = MPSGraphResizeBilinear; break;
+    case GridSamplerInterpolation::Nearest:
+      samplingMode = MPSGraphResizeNearest; break;
+    case GridSamplerInterpolation::Bicubic:
+      TORCH_CHECK(false, "MPS: Unsupported Bicubic interpolation"); break;
+    default:
+      TORCH_CHECK(false, "MPS: Unrecognised interpolation mode: ", interpolation_mode); break;
+   }
+
+  MPSStream *stream = getCurrentMPSStream();
+
+  struct CachedGraph : public MPSCachedGraph
+  {
+    CachedGraph(MPSGraph *graph) : MPSCachedGraph(graph) {}
+    MPSGraphTensor* inputTensor_ = nil;
+    MPSGraphTensor* gridTensor_ = nil;
+    MPSGraphTensor* outputTensor_ = nil;
+  };
+
+  MPSGraphCache* cache_ = MPSGraphCache::getInstance();
+
+  @autoreleasepool {
+    string key = "grid_sampler_2d_mps"                     +
+                  getTensorsStringKey({input, grid})       +
+                  ":" + std::to_string(interpolation_mode) +
+                  ":" + std::to_string(padding_mode)       +
+                  ":" + std::to_string(align_corners);
+
+    CachedGraph* cachedGraph = static_cast<CachedGraph *>(cache_->LookUp(key));
+    if(!cachedGraph) {
+      MPSCachedGraph *tmpCachedGraph = cache_->CreateCachedGraph(key, ^ MPSCachedGraph * () {
+
+        CachedGraph *newCachedGraph = nil;
+        @autoreleasepool {
+          MPSGraph* mpsGraph = make_mps_graph();
+          newCachedGraph = new CachedGraph(mpsGraph);
+
+          MPSGraphTensor* inputTensor = mpsGraphRankedPlaceHolder(mpsGraph, input);
+          MPSGraphTensor* gridTensor = mpsGraphRankedPlaceHolder(mpsGraph, grid);
+
+          MPSGraphTensor* outputTensor = [mpsGraph sampleGridWithSourceTensor: inputTensor
```

Comment:
@DenisVieriu97 , you will need to add check for `respondsToSelector` here. We can use Ventura 13.1 or something, but i would rather make this Instance specific ...